File: 704-binary-search/704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int low = 0;
        int high = nums.length-1;
        
        return binary_search(low, high, target, nums);
    }
    
    int binary_search(int low, int high, int target, int[] nums) {
        if (low > high) return -1;
        int mid = low + (high-low) / 2;
        if (nums[mid] < target) return binary_search(mid+1, high, target, nums);
        else if (nums[mid] > target) return binary_search(low, mid-1, target, nums); 
        return mid;
    }
    
    
}

File: 704-binary-search/704-binary-search.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        low, high = 0, len(nums)-1
        
        while low <= high:
            mid = low + ((high - low) // 2)
            print(nums[mid])
            if nums[mid] == target: return mid
            elif nums[mid] < target: low = mid + 1
            else: high = mid - 1
            
        return -1
        